src/app/store/submitted-transactions/submitted-transactions.slice.ts

```diff
@@ -0,0 +1,22 @@
+import { createEntityAdapter, createSlice, EntityId, PayloadAction } from '@reduxjs/toolkit';
+
+export type SubmittedTransaction = { rawTx: string; txId: string };
+
+const submittedTransactionsAdapter = createEntityAdapter<SubmittedTransaction>({
+  selectId: submittedTransaction => submittedTransaction.txId,
+});
+
+export const initialSubmittedTransactionsState = submittedTransactionsAdapter.getInitialState();
+
+export const submittedTransactionsSlice = createSlice({
+  name: 'submittedTransactions',
+  initialState: initialSubmittedTransactionsState,
+  reducers: {
+    transactionAdded(state, action: PayloadAction<SubmittedTransaction>) {
+      submittedTransactionsAdapter.addOne(state, action.payload);
+    },
+    transactionRemoved(state, action: PayloadAction<EntityId>) {
+      submittedTransactionsAdapter.removeOne(state, action.payload);
+    },
```

Comment:
This action names describe the result we want from an action or event. Per https://redux.js.org/style-guide/#model-actions-as-events-not-setters better to describe the event itself.

`newTransactionSubmitted` or `transactionBroadcastSuccessfully`, and `transactionEnteredMempool`?

Reply:
Ok, again, was referring to the example in docs.

`bookAdded: booksAdapter.addOne`

Reply:
Sure, I see your thinking. I guess in some simple cases, the event is pretty much is the setter: If I "add a book to my collection" that maps 1:1 with adding a book to the state.

Here adding a submitted transaction to the result we want to happen after successfully broadcasting a transaction. If we were to have a broadcast tx action, we might have even reused that.